pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -985,10 +965,23 @@ func (b *MasqueradeBindMechanism) createBridge() error {
 }
 
 func (b *MasqueradeBindMechanism) createNatRules(protocol iptables.Protocol) error {
-	if Handler.HasNatIptables(protocol) {
+	err := Handler.ConfigureIpForwarding(protocol)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to configure ip forwarding")
+		return err
+	}
+
+	ipString := "ipv4"
+	if protocol == iptables.ProtocolIPv6 {
+		ipString = "ipv6"
+	}
+
+	if Handler.NftablesLoad(fmt.Sprintf("%s-nat", ipString)) == nil {
```

Comment:
Is not very symmetric, now we identify protocol inside of `ConfigureIpForward` but we still need to do so to check if we have to use iptables or nftables, let's move the protocol checking inside Nftables functions too adding a new helper to `Handler`.

```golang
func HasNatNftables(string protocol) bool {
        ipString := "ipv4"
	if protocol == iptables.ProtocolIPv6 {
		ipString = "ipv6"
	}
       err := Handler.NftablesLoad(fmt....)
       return err == nil       
}
```

so this will be simpler and symmetric.

```golang
if Handler.HasNatNftables(protocol) {
....
} else if Handler.HasNatIptables(protocl) {
....
}
...
```

Reply:
The same asymmetry existed before this PR.
I don't want to move `Handler.NftablesLoad` to `HasNatNftables` since when invoking `HasNatNftables` I would expect just a check to be performed. And in this case, the table will be actually loaded.

Reply:
I mean adding new function `HasNatNftables` that will call `NftablesLoad` not moving also the ipv6 vs ipv4 was not there before the PR at this function.

Reply:
I think that `HasNatNftables` invoking `NftablesLoad` is misleading.
A method that starts with `Has`, shouldn't load anything, just check the existing state.

Reply:
Then call it "LoadNatNftables" and do the protocol checking and name composing there, what do you think ?

Reply:
+1 on not adding a side effect into the function - loading the table.

+1 on changing the function signature to `LoadNATNftables(protocol iptables.Protocol) error` and move all the type translation to it.

Reply:
cnahged `NftablesLoad` to receive the protocol as a parameter. The `fnName` composing will be done inside the method.